src/helpers/BasicPointsAndBadges/ConnectExternalAccountActivity.ts

```diff
@@ -1,18 +1,26 @@
 import MyDataHelps from "@careevolution/mydatahelps-js";
 import type { BasicPointsForBadgesActivity, BasicPointsForBadgesActivityState } from "./Activities";
 
+export type ConnectExternalActivityType = "Provider" | "Health Plan" | "Device Manufacturer";
+
 export interface ConnectExternalAccountActivity extends BasicPointsForBadgesActivity {
     type: "connectExternalAccount";
-    providerCategories: string[];
+    providerCategories: ConnectExternalActivityType[];
 }
 
 interface ConnectExternalAccountActivityState extends BasicPointsForBadgesActivityState {
     providersConnected?: number[];
 }
 
 export async function awardConnectExternalAccountActivityPoints(activity: ConnectExternalAccountActivity, activityState: ConnectExternalAccountActivityState) {
+    //convert to array to protect against new types added in consumers but not yet here
+    const activityProviderCategories = activity.providerCategories as string[]; 
+
+    //Get the external accounts
     const connectedAccounts = await MyDataHelps.getExternalAccounts();
-    const allProviders = connectedAccounts.filter(account => activity.providerCategories.includes(account.provider.category)).map(t => t.id);
+    //Filter for the accounts that match this activity. Return the external account id
```

Comment:
I think these are probably simple enough operations that the comments probably aren't necessary.  I also tend to feel like comments get out of date fairly easily when adjustments or refactorings occur, so they often can't be relied upon.

If some form of inline documentation is desirable, I think a slightly less brittle way to do it is to create helper functions that clearly describe what is happening with the function name.  For example, something like:

```
function filterAccountsByActivity(connectedAccounts: ExternalAccount[], activity: ConnectExternalAccountActivity) : number[] {
    return connectedAccounts.filter(account => activity.providerCategories.includes(account.provider.category as ConnectExternalActivityType)).map(account => account.id);
}
```

It is probably splitting hairs, but I think the function name is more likely to be updated if the functionality changes than the comment.

Of course, the best way to document what is/should be happening is with the unit tests you've created.  😃

Reply:
I agree. Especially with the change to map provider id and ut around it.